.github/workflows/_test_requiring_torch_cuda.yml

```diff
@@ -0,0 +1,129 @@
+        default: false
+      timeout-minutes:
+        required: false
+        type: number
+        default: 270
+        description: |
+          Set the maximum (in minutes) how long the workflow should take to finish
+            timeout-minutes:
+
+    secrets:
+      gcloud-service-key:
+        required: true
+        description: Secret to access Bazel build cache
+jobs:
+  test:
+    runs-on: ${{ inputs.runner }}
+    container:
+      image: ${{ inputs.dev-image }}
+      options: "--gpus all --shm-size 16g"
+    timeout-minutes: ${{ inputs.timeout-minutes }}
+    env:
+      GCLOUD_SERVICE_KEY: ${{ secrets.gcloud-service-key }}
+      GOOGLE_APPLICATION_CREDENTIALS: /tmp/default_credentials.json
+      USE_COVERAGE: ${{ inputs.collect-coverage && '1' || '0' }}
+      BAZEL_JOBS: 16
+      BAZEL_REMOTE_CACHE: 1
+    steps:
+      # See https://github.com/actions/checkout/issues/1014#issuecomment-1906802802
+      # TODO: need to find a way to reuse these steps.
+      - name: Clean up workspace
+        run: |
+          ls -la
+          rm -rvf ${GITHUB_WORKSPACE}/*
+      - name: Setup gcloud
+        shell: bash
+        run: |
+          echo "${GCLOUD_SERVICE_KEY}" > $GOOGLE_APPLICATION_CREDENTIALS
+      - name: Fetch torch/torch_xla/torchvision wheels
+        uses: actions/download-artifact@v4
+        with:
+          name: torch-xla-wheels
+          path: /tmp/wheels/
+          pattern: torch-*.whl 
+      - name: Remove torch wheel
+        shell: bash
+        run: |
+          rm -rf /tmp/wheels/torch-*
+      - name: Fetch torch/torch_xla/torchvision wheels with CUDA enabled
+        uses: actions/download-artifact@v4
+        with:
+          name: torch-with-cuda-xla-with-cuda-wheels
+          path: /tmp/wheels/
+          pattern: torch-*.whl 
+      - name: Fetch CUDA plugin
+        uses: actions/download-artifact@v4
+        with:
+          name: cuda-plugin
+          path: /tmp/wheels/
+      - name: Setup CUDA environment
+        shell: bash
+        run: |
+          # TODO: Make PJRT_DEVICE=CPU work with XLA_REGISTER_INSTALLED_PLUGINS=1
+          echo "XLA_REGISTER_INSTALLED_PLUGINS=1" >> $GITHUB_ENV
+
+          echo "PATH=$PATH:/usr/local/cuda-12.1/bin" >> $GITHUB_ENV
+          echo "LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/cuda-12.1/lib64" >> $GITHUB_ENV
+      - name: Check GPU
+        run: nvidia-smi
+      - name: Install wheels
+        shell: bash
+        run: |
+          pip install /tmp/wheels/*.whl
+          # TODO: Add these in setup.py
+          pip install fsspec
+          pip install rich
+
+          echo "Import check..."
+          python -c "import torch, torch_xla, torchvision"
+          echo "Import check done."
+          echo "Check if CUDA is available for PyTorch..."
+          python -c "import torch; assert torch.cuda.is_available()"
+          echo "CUDA is available for PyTorch."
+      - name: Record PyTorch commit
```

Comment:
Just be aware, there's a small chance of version skew in this workflow. You could end up with a broken workflow if there's a breaking change between when `_bulid_torch_xla` and `_build_torch_with_cuda` launch:

- _build_torch_xla launches, pulls pytorch, etc
- breaking change merged to pytorch
- _build_torch_with_cuda, pulls pytorch, etc

It's up to you if you want to find a way to fix that now or just wait to see if it's an issue in practice. There should really only a window of seconds between when workflow launch.

The permanent fix would probably be to add a parent workflow step that "records" the PyTorch commit hash before either of the builds run. Maybe something to address for whoever factors out the setup steps.

Reply:
It's great point. I'll see if it's an issue in practice. If so, I'll fix it accordingly.

Reply:
Actually I saw it in this PR:
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/lib/python3.10/site-packages/torch_xla/__init__.py", line 7, in <module>
    import _XLAC
ImportError: /usr/local/lib/python3.10/site-packages/_XLAC.cpython-310-x86_64-linux-gnu.so: undefined symbol: _ZN3c[101](https://github.com/pytorch/xla/actions/runs/9308352858/job/25626741718?pr=7140#step:11:102)5SmallVectorBaseIjE8grow_podEPvmm
```
suggesting the torch wheel and torch_xla wheel are not compatible.

So I added a parent job to get the commit and pass it to the reusable workflow `build-torch-xla` and `build-torch-with-cuda` according to https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-outputs-from-a-reusable-workflow